Sources/Apodini/Semantic Model Builder/Model/TypeReflection/EnrichedInfo.swift

```diff
@@ -21,6 +23,50 @@ struct EnrichedInfo {
     var cardinality: Cardinality = .exactlyOne
 }
 
+extension EnrichedInfo.Cardinality: Equatable {
+    public static func == (lhs: Self, rhs: Self) -> Bool {
+        switch (lhs, rhs) {
+        case (.zeroToOne, .zeroToOne), (.exactlyOne, .exactlyOne):
+            return true
+        case let (.zeroToMany(lhsCollection), .zeroToMany(rhsCollection)):
+            return (lhsCollection) == (rhsCollection)
+        default:
+            return false
+        }
+    }
+}
+
+extension EnrichedInfo.CollectionContext: Equatable {
+    public static func == (lhs: Self, rhs: Self) -> Bool {
+        switch (lhs, rhs) {
+        case (.array, .array):
+            return true
+        case let (.dictionary(lhsKey, lhsValue), .dictionary(rhsKey, rhsValue)):
+            return (lhsKey, lhsValue) == (rhsKey, rhsValue)
+        default:
+            return false
+        }
+    }
+}
+
+extension EnrichedInfo: Equatable {
+    public static func == (lhs: EnrichedInfo, rhs: EnrichedInfo) -> Bool {
+        if lhs.typeInfo.mangledName != rhs.typeInfo.mangledName {
```

Comment:
```suggestion
        lhs.typeInfo.type == rhs.typeInfo.type
            && lhs.propertyInfo?.name == rhs.propertyInfo?.name
            && lhs.propertiesOffset == rhs.propertiesOffset
            && lhs.cardinality == rhs.cardinality
```

Plus, I exchanged `typeInfo.mangledName` for `typeInfo.type`.

And I'd put `EnrichedInfo: Equatable` above the other two extensions, preserving the hierarchy.
Futhermore, I'd put all three of them at the end of the file, maybe below a a `MARK` comment:
`// MARK: - EnrichedInfo: Equatable`.

Reply:
thanks, very good suggestions, solved via 978a99b